client/db/bolt/db.go

```diff
@@ -1624,6 +1624,129 @@ func (idx *timeIndexNewest) add(t uint64, k []byte, b *bbolt.Bucket) {
 	})
 }
 
+// DeleteInactiveOrders deletes orders that no longer needed for normal
+// operations. Optionally accepts a time to delete orders with a later time
+// stamp.
+func (db *BoltDB) DeleteInactiveOrders(olderThan *time.Time) (deletedOrders []*dexdb.MetaOrder, finished bool, err error) {
```

Comment:
While it is logical to have the delete and backup operations be facilitated by the same db method by returning a slice of deleted orders, I'm fairly certain that is going to be a ton of memory in some notable use cases.

A trick to consider is to pass a function to `DeleteInactiveOrders` like

```go
func (db *BoltDB) DeleteInactiveOrders(olderThan *time.Time, removed func(*dexdb.MetaOrder) error) (bool, error)
```

A caller like `(*Core).DeleteArchivedRecords` would then handle them in a stream.  A nil `removed` function could be an option if the caller does not want them.

I had to do something similar with `MarketMatchesStreaming` server-side.  See `server/admin.(*Server).apiMarketMatches` where it streams `MatchData` into JSON.

Reply:
Will try it with the function pass.

Reply:
Passing a function.